Detectors/DCS/testWorkflow/DCSDataProcessorSpec.h

```diff
@@ -0,0 +1,244 @@
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#ifndef O2_DCS_DATAPROCESSOR_H
+#define O2_DCS_DATAPROCESSOR_H
+
+/// @file   DataGeneratorSpec.h
+/// @brief  Dummy data generator
+
+#include <unistd.h>
+#include <TRandom.h>
+#include <TStopwatch.h>
+#include "DetectorsDCS/DataPointIdentifier.h"
+#include "DetectorsDCS/DataPointValue.h"
+#include "DetectorsDCS/DataPointCompositeObject.h"
+#include "DetectorsDCS/DeliveryType.h"
+#include "DetectorsDCS/DCSProcessor.h"
+#include "DetectorsCalibration/Utils.h"
+#include "CCDB/CcdbApi.h"
+#include "Framework/DeviceSpec.h"
+#include "Framework/ConfigParamRegistry.h"
+#include "Framework/ControlService.h"
+#include "Framework/WorkflowSpec.h"
+#include "Framework/Task.h"
+#include "Framework/Logger.h"
+
+namespace o2
+{
+namespace dcs
+{
+
+using namespace o2::dcs;
+using DPID = o2::dcs::DataPointIdentifier;
+using DPVAL = o2::dcs::DataPointValue;
+using DPCOM = o2::dcs::DataPointCompositeObject;
+
+class DCSDataProcessor : public o2::framework::Task
+{
+ public:
+  enum Detectors {
+    kTest,
+    kTPC,
+    kNdetectors
+  };
+
+  void init(o2::framework::InitContext& ic) final
+  {
+    std::vector<DPID> aliasVect;
+
+    DPID dpidtmp;
+    DeliveryType typechar = RAW_CHAR;
+    std::string dpAliaschar = "TestChar_0";
+    DPID::FILL(dpidtmp, dpAliaschar, typechar);
+    aliasVect.push_back(dpidtmp);
+
+    std::vector<int> vectDet{kTest, kTPC};
+    mDetectorAlias[dpidtmp] = vectDet;
+
+    DeliveryType typeint = RAW_INT;
+    for (int i = 0; i < 50000; i++) {
+      std::string dpAliasint = "TestInt_" + std::to_string(i);
+      DPID::FILL(dpidtmp, dpAliasint, typeint);
+      aliasVect.push_back(dpidtmp);
+      mDetectorAlias[dpidtmp] = vectDet;
+    }
+
+    DeliveryType typedouble = RAW_DOUBLE;
+    for (int i = 0; i < 4; i++) {
+      std::string dpAliasdouble = "TestDouble_" + std::to_string(i);
+      DPID::FILL(dpidtmp, dpAliasdouble, typedouble);
+      aliasVect.push_back(dpidtmp);
+      mDetectorAlias[dpidtmp] = vectDet;
+    }
+
+    DeliveryType typestring = RAW_STRING;
+    std::string dpAliasstring0 = "TestString_0";
+    DPID::FILL(dpidtmp, dpAliasstring0, typestring);
+    aliasVect.push_back(dpidtmp);
+    mDetectorAlias[dpidtmp] = vectDet;
+
+    mDCSproc.init(aliasVect);
+    mDCSproc.setMaxCyclesNoFullMap(ic.options().get<int64_t>("max-cycles-no-full-map"));
+    mDCSproc.setName("Test0Det");
+    mDCSprocVect[kTest].init(aliasVect);
+    mDCSprocVect[kTest].setMaxCyclesNoFullMap(ic.options().get<int64_t>("max-cycles-no-full-map"));
+    mDCSprocVect[kTest].setName("Test1Det");
+    mProcessFullDeltaMap = ic.options().get<bool>("process-full-delta-map");
+  }
+
+  void run(o2::framework::ProcessingContext& pc) final
+  {
+    auto tfid = o2::header::get<o2::framework::DataProcessingHeader*>(pc.inputs().get("input").header)->startTime;
+    mDCSproc.setTF(tfid);
```

Comment:
here you use ``mDCSproc``, but then  actual data filling and ccdb preparation is done on mDCSprocVect

Reply:
I forgot to set it for mDCSprocVect. Fixed. 
For clarity: mDCSproc is there to benchmark in case we do not process per timestamp.